core/tee/tee_svc_cryp.c

```diff
@@ -517,18 +517,26 @@ static void op_u32_to_binary_helper(uint32_t v, uint8_t *data,
 				    size_t data_len, size_t *offs)
 {
 	uint32_t field;
+	size_t temp;
+
+	if (ADD_OVERFLOW(*offs, sizeof(field), &temp))
```

Comment:
In practice this can't happen as we're dealing with buffers in secure memory and `*offs` always starts from `0`.
However if we still want to check for this then we should do something useful when an error is detected, in this case return `false` (we'd need to change the type of the function etc too of course) to allow the caller to do some error handling.

I'd rather use the name `next_offs` instead of `temp`.

Reply:
First i will change the prototype of the function from void to bool returned,  and  maybe the caller should deal with the return value,  and the naming is ok.